### ISSUE TITLE
pxelinux doesn't support UEFI

### ISSUE THREAD

**Issue by peterhoeg** [NixOS/nixpkgs#41403]
## Issue description

pxelinux (from syslinux) is compiled without support for UEFI PXE boot: https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/syslinux/default.nix#L47

I had a quick look at getting it to compile uefi support but I didn't get very far.

## Technical details

 - system: `"x86_64-linux"`
 - host os: `Linux 4.16.11, NixOS, 18.09.git.c2a4a5143f2M (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0.2`
 - channels(root): `""`
 - channels(peter): `""`
 - nixpkgs: `/home/peter/src/active/nixpkgs_unstable`

**Comment by markuskowa** [comment]
Does the NixOS netboot image support UEFI boot?  I did try netbooting NixOS (18.03) a few weeks ago with a pre-compiled iPXE UEFI loader. It started booting the kernel but failed to mount the initrd (kernel panic no root).

**Comment by peterhoeg** [comment]
The image is using pxelinux.0 instead of lpxelinux.0 it seems and you need lpxelinux for HTTP support so maybe that's what is going on.

**Comment by markuskowa** [comment]
The http support seemed to worked just fine in my case. I did serve  the iPXE UEFI loader via tftp, the kernel image and the initrd via http (initrd is too big to load trouble-free via tftp). The iPXE loader pulled bzImage and initrd via http and began booting the kernel. The kernel then failed to mount the ramdisk for some reason I do not understand.
I solved the problem by setting 'Legacy mode' in the firmware (Supermicro) and booting everything in classical BIOS/PXE mode (no UEFI components involved). I can try to reproduce the problem and send you some screenshots if that helps. 
My vague guess is that the NixOS netboot image needs some modifications to be UEFI capable. 

I just wanted to raise awareness that there are potential problems with UEFI-PXE down the road.
